src/App.js

```diff
@@ -1,13 +1,112 @@
-import React from 'react';
+import React from "react";
+import TodoList from "./components/TodoComponents/TodoList";
+import TodoForm from "./components/TodoComponents/TodoForm";
+import uuid from "uuid";
+
+const initialTodos = [
+  {
+    task: "Organize Garage",
+    id: uuid(),
+    completed: true
+  },
+  {
+    task: "Bake Cookies",
+    id: uuid(),
+    completed: false
+  }
+];
+
+const fakeTodosEndpoint = () => {
+  return Promise.resolve(initialTodos);
+};
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.state = {
+      todos: [],
+      newTodoValue: ""
+    };
+  }
+
+  componentDidMount() {
+    fakeTodosEndpoint().then(todos => {
+      this.setState(oldState => {
+        return {
+          todos
+        };
+      });
+    });
+  }
+
+  valueChange = event => {
+    const newValue = event.target.value;
+    console.log(event.target.value);
+    this.setState(oldState => {
+      return {
+        newTodoValue: newValue
+      };
+    });
+  };
+
+  addTodo = event => {
+    console.log("click");
+    event.target.value = "";
+    this.setState(oldState => {
+      return {
+        newTodoValue: "",
+        todos: oldState.todos.concat({
+          completed: false,
+          id: uuid(),
+          task: oldState.newTodoValue
+        })
+      };
+    });
+  };
+
+  markCompleted = id => {
+    
+    this.setState(oldState => {
+      return {
+        todos: oldState.todos.map(todo => {
+          
+          if (todo.id === id) {
+            return {
+              ...todo,
+              completed: !todo.completed,
+            };
+          } 
+          return todo;
+          
+        })
+      };
+    });
+  };
+
+  clearCompleted = event => {
+    this.setState(oldState => {
+      return {
+        todos: oldState.todos.filter(todo => todo.completed === false)
+      };
+    });
+  };
+
+  //onClick={this.markCompleted(todo.id)}
+
   render() {
     return (
       <div>
-        <h2>Welcome to your Todo App!</h2>
+        <TodoList
+        todos={this.state.todos}
+        markCompleted={this.markCompleted}
+        />
+
+        <TodoForm
```

Comment:
Great job with the project! To make your input empty any time you submit a new todo, send the `newTodoValue` in state as props to the `TodoForm`component. Like this => 


```
<TodoForm
    **val={this.state.newTodoValue}**
    todos={this.state.todos}
    clearCompleted={this.clearCompleted}
    addTodo={this.addTodo}
    valueChange={this.valueChange}
 />
```